core/src/main/java/com/amplifyframework/analytics/AnalyticsCategory.java

```diff
@@ -94,13 +94,17 @@ public void recordEvent(@NonNull final AnalyticsEvent analyticsEvent)
     }
 
     @Override
-    public void registerGlobalProperties(@NonNull Properties properties) {
-        throw new UnsupportedOperationException("This operation is currently not supported.");
+    public void registerGlobalProperties(@NonNull Properties properties) throws AnalyticsException {
+        if (enabled) {
```

Comment:
Will there be any error reported if analytics is not enabled?

Reply:
No we don't report an error currently. Analytics APIs are a no-op if Analytics is disable. I think this allows them to conditionally enable and disable calls without error being reported.